bc_obps/service/report_service.py

```diff
@@ -140,30 +136,22 @@ def save_facility_report(cls, report_version_id: int, facility_id: int, data: Fa
         Returns:
             FacilityReport: The updated or created FacilityReport instance.
         """
-        try:
-            # Fetch or create a FacilityReport instance
-            facility_report, _ = FacilityReport.objects.update_or_create(
-                id=facility_id,
-                report_version__id=report_version_id,
-                defaults={
-                    'facility_name': data.facility_name.strip(),
-                    'facility_type': data.facility_type.strip(),
-                    'facility_bcghgid': data.facility_bcghgid.strip(),
-                },
-            )
-
-            # Update ManyToMany fields (activities)
-            if data.activities:
-                facility_report.activities.set(Activity.objects.filter(id__in=data.activities))
+        # Fetch or create a FacilityReport instance
+        facility_report, _ = FacilityReport.objects.update_or_create(
+            id=facility_id,
```

Comment:
Same, this will never update, we're mixing primary keys

Reply:
Same thing as above comment, I fixed the uuid - id mismatch here & changed this to a strict update in #2223. We shouldn't need a create or update here as the facility report is created on [report create](https://github.com/bcgov/cas-registration/blob/develop/bc_obps/service/report_service.py#L66), we should expect this to exist already by the time this endpoint is hit